backend/database.js

```diff
@@ -51,6 +51,39 @@ export default (sbp('sbp/selectors/register', {
       }
     })
   },
+  'backend/db/streamEntriesBefore': async function (before: string, limit: number): Promise<*> {
+    let prefix = '['
+    let currentHEAD = before
+    let entry = await sbp('chelonia/db/getEntry', currentHEAD)
+    if (!entry) {
+      throw Boom.notFound(`entry ${currentHEAD} doesn't exist!`)
+    }
+    limit++ // to return `before` apart from the `limit` number of events
+    // NOTE: if this ever stops working you can also try Readable.from():
+    // https://nodejs.org/api/stream.html#stream_stream_readable_from_iterable_options
+    return new Readable({
+      async read (): any {
+        try {
+          if (!currentHEAD || !limit) {
+            this.push(']')
+            this.push(null)
+          } else {
+            entry = await sbp('chelonia/db/getEntry', currentHEAD)
+            const json = `"${strToB64(entry.serialize())}"`
+            this.push(prefix + json)
+            prefix = ','
+
+            limit--
+            currentHEAD = entry.message().previousHEAD
+          }
+        } catch (e) {
+          console.error(`read(): ${e.message}:`, e)
```

Comment:
Could you add above line 80 the following comment?

```js
// TODO: properly return an error to caller, see https://nodejs.org/api/stream.html#errors-while-reading
```

Reply:
Sure. Why not.